### PR TITLE
Neldermead initial simplex

### PR DESCRIPTION
**Pull request by pv**
Yet another PR for specifying initial simplex for nelder-mead.
Takes the parameter form from gh-3729 and implementation from gh-3634.
See gh-5856 for discussion.

### CODE REVIEW

File: scipy/optimize/optimize.py [hunk 1]
```diff
@@ -327,6 +327,8 @@ def fmin(func, x0, args=(), xtol=1e-4, ftol=1e-4, maxiter=None, maxfun=None,
         Set to True to print convergence messages.
     retall : bool, optional
         Set to True to return list of solutions at each iteration.
+    initial_simplex : array_like of shape (N + 1, N), optional
```
Comment (andyfaff) [inline scipy/optimize/optimize.py]: It's not explicitly mentioned in this docstring what N is, although one can tell from the code. Is it worth giving a link to how to make a simplex?

**Comment by pv** [comment]
Updated.

**Comment by BenFrantzDale** [comment]
I'm psyched to see someone finally implementing a workable version of this!

**Comment by pv** [comment]
Updated.

File: scipy/optimize/optimize.py [hunk 2]
```diff
@@ -421,37 +432,48 @@ def _minimize_neldermead(func, x0, args=(), callback=None,
...
     rho = 1
     chi = 2
     psi = 0.5
     sigma = 0.5
-    one2np1 = list(range(1, N + 1))
+    nonzdelt = 0.05
+    zdelt = 0.00025
+
+    if initial_simplex is None:
+        x0 = asfarray(x0).flatten()
```
Comment (Eric89GXL) [inline scipy/optimize/optimize.py]: You could move this outside the `if` clause, then you wouldn't need to do a very similar cast in the `else` clause below.

File: scipy/optimize/optimize.py [hunk 3]
```diff
@@ -421,37 +432,48 @@ def _minimize_neldermead(func, x0, args=(), callback=None,
...
+                y[k] = (1 + nonzdelt)*y[k]
+            else:
+                y[k] = zdelt
+            sim[k + 1] = y
+    else:
+        sim = np.asfarray(initial_simplex).copy()
+        if sim.ndim != 2 or sim.shape[0] != sim.shape[1] + 1:
+            raise ValueError("`initial_simplex` should be an array of shape (N+1,N)")
+        if np.asarray(x0).size != sim.shape[1]:
+            warnings.warn("Size of `initial_simplex` is not consistent with `x0`")
```
Comment (Eric89GXL) [inline scipy/optimize/optimize.py]: Should we raise an error instead? Or are there enough use cases where people will do this intentionally to permit it?

**Comment by Eric89GXL** [comment]
Otherwise LGTM

**Comment by pv** [comment]
Updated

**Comment by andyfaff** [comment]
I'll merge this tomorrow unless you have extra changes to make. Thanks @pv, @isjoung

**Comment by BenFrantzDale** [comment]
Yay!

**Comment by AdriaanRol** [comment]
:+1:

**Pull request merged**